atom/browser/mac/atom_application_delegate.mm

```diff
@@ -52,9 +52,11 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication*)sender
 - (BOOL)applicationShouldHandleReopen:(NSApplication*)theApplication
                     hasVisibleWindows:(BOOL)flag {
   atom::Browser* browser = atom::Browser::Get();
+  browser->Activate(static_cast<bool>(flag));
   if (flag) {
     return YES;
   } else {
+    // Deprecated API, for compatibility.
     browser->ActivateWithNoOpenWindows();
```

Comment:
We can remove the whole `OnActivateWithNoOpenWindows` thing in C++ and write a small compatibility layer for `activate-with-no-open-windows` in JavaScript.